src/arviz_stats/base/dataarray.py

```diff
@@ -402,5 +402,19 @@ def autocorr(self, da, dim=None):
             output_core_dims=[dims],
         )
 
+    def mode(self, da, dim=None):
+        """Compute mode on DataArray input."""
+        if dim is None:
+            dims = list(da.dims)
+        else:
+            dims = dim if isinstance(dim, (list | tuple)) else [dim]
```

Comment:
I am not completely sure we want this as the behaviour. I see the appeal of matching the behaviour of `mean` and `median`, but at the same time we are also making this be the odd one out within `azstats` accessors and I am not sure where to draw the line. Why does `eti` behave differently than quantile?

disclaimer: I have a conflict of interest because I like arviz default much more than xarray one, so if this is the default I will be constantly writing `dim=["chain", "draw"]` and I get intrusive thoughts every few months about adding a `ds.azstats.mean()` that only does `ds.mean(rcParams["sample_dims"])`

Reply:
It could be a good idea to have an arviz's version of mean and median, but stop there! haha.